LoopKitUI/Views/Settings Editors/TherapySettingsView.swift

```diff
@@ -372,45 +378,131 @@ struct SectionWithTapToEdit<Content, NavigationDestination>: View where Content:
+                            back()
+                        },
+                        mode: self.viewModel.mode
+                    ))
+                }
+            }
         case .basalRate:
-            return AnyView(BasalRatesReview(mode: viewModel.mode, viewModel: viewModel))
+            return { back in
+                AnyView(BasalRateScheduleEditor(
+                    schedule: self.viewModel.therapySettings.basalRateSchedule,
+                    supportedBasalRates: self.viewModel.pumpSupportedIncrements!.basalRates ,
+                    maximumBasalRate: self.viewModel.therapySettings.maximumBasalRatePerHour,
+                    maximumScheduleEntryCount: self.viewModel.pumpSupportedIncrements!.maximumBasalScheduleEntryCount,
+                    syncSchedule: self.viewModel.syncPumpSchedule,
+                    onSave: { newRates in
+                        self.viewModel.saveBasalRates(basalRates: newRates)
+                        back()
+                    },
+                    mode: self.viewModel.mode
+                ))
+            }
         case .deliveryLimits:
-            return AnyView(DeliveryLimitsReview(mode: viewModel.mode, viewModel: viewModel))
+            return { back in
+                AnyView(DeliveryLimitsEditor(
+                    value: self.viewModel.deliveryLimits,
+                    supportedBasalRates: self.viewModel.pumpSupportedIncrements!.basalRates,
+                    scheduledBasalRange: self.viewModel.therapySettings.basalRateSchedule?.valueRange(),
+                    supportedBolusVolumes: self.viewModel.pumpSupportedIncrements!.bolusVolumes,
+                    onSave: { limits in
+                        self.viewModel.saveDeliveryLimits(limits: limits)
+                        back()
+                    },
+                    mode: self.viewModel.mode
+                ))
+            }
         case .insulinModel:
-            if viewModel.therapySettings.glucoseUnit != nil && viewModel.therapySettings.insulinModelSettings != nil && viewModel.therapySettings.insulinSensitivitySchedule != nil {
-                return AnyView(InsulinModelSelection(viewModel: viewModel.insulinModelSelectionViewModel,
-                                                     glucoseUnit: self.viewModel.therapySettings.glucoseUnit!,
-                                                     supportedModelSettings: viewModel.supportedInsulinModelSettings,
-                                                     appName: viewModel.appName,
-                                                     mode: viewModel.mode))
+            if self.viewModel.therapySettings.glucoseUnit != nil && self.viewModel.therapySettings.insulinModelSettings != nil && self.viewModel.therapySettings.insulinSensitivitySchedule != nil {
```

Comment:
One concern I have with making these conditionally available is that it would be impossible to edit this settings if these things weren't set - could it make more sense to provide a sensible default if there isn't one already? This is similar to the behavior of the current settings editors

Reply:
I'll make sure it doesn't crash for this PR.  Providing sensible default behavior really should involve design.